### DISCUSSION TITLE
How to get all variables from an expression

### DISCUSSION THREAD

**Discussion by star2478** [antonmedv/expr#743] [Q&A] [answered]
Hi
How to get all variables from parsing an expression? For example
* an expression: 
```
let arr = filter(user_list, {.user_id == "xxxx"}); 
len(arr)>0? (arr[0].mobile=="123"?arr[0].employee:(arr[0].mobile=="13800001111"?arr[0].employee2:"0") ) :-1
```
* all variables: user_list, user_id, mobile, employee, employee2

**Comment by antonmedv** [comment, marked as answer]
Here is a good example: https://expr-lang.org/docs/visitor
